util/retry/retry.go

```diff
@@ -19,24 +19,17 @@ package retry
 
 import (
 	"fmt"
-	"math/rand"
 	"time"
 
+	"github.com/cenkalti/backoff"
+
 	"github.com/cockroachdb/cockroach/util"
 	"github.com/cockroachdb/cockroach/util/log"
 )
 
-// retryJitter specifies random jitter to add to backoff
-// durations. Specified as a percentage of the backoff.
-const retryJitter = 0.15
-
-// Status is an enum describing the possible statuses of a
-// backoff / retry worker function.
-type Status int32
-
 // MaxAttemptsError indicates max attempts were exceeded.
 type MaxAttemptsError struct {
-	MaxAttempts int
+	MaxAttempts uint
```

Comment:
Understood the change wasn't about the range of `int` vs `uint`. While it doesn't make sense for `MaxAttempts` to be negative, it isn't harmful for this field to be an `int`. Making it a `uint` can make interaction with other types more irritating. While I doubt that would be the case for this particular field, it is a general blanket statement that we should be preferring signed integer types where possible. 

PS Preferring signed integer types is also the go-way. `len(foo)` returns an `int`, not a `uint`.